tools/torchci/clickhouse.py

```diff
@@ -0,0 +1,49 @@
+import json
+import os
+from functools import lru_cache
+from typing import Any, Dict
+
+import clickhouse_connect
+from torchci.utils import REPO_ROOT
+
+
+@lru_cache(maxsize=1)
+def get_clickhouse_client() -> Any:
+    endpoint = os.environ["CLICKHOUSE_ENDPOINT"]
+    # I cannot figure out why these values aren't being handled automatically
+    # when it is fine in the lambda
+    if endpoint.startswith("https://"):
```

Comment:
You might want to use something like https://docs.python.org/3/library/urllib.parse.html to get the hostname

Reply:
I did some testing right now, and it doesn't seem to handle having vs not having https:// well

I also found this in the docs
"Following the syntax specifications in [RFC 1808](https://datatracker.ietf.org/doc/html/rfc1808.html), urlparse recognizes a netloc only if it is properly introduced by ‘//’. Otherwise the input is presumed to be a relative URL and thus to start with a path component."